File: analyses/analysis125.R
library(dplyr)
library(data.table)

# --- Analysis at row 125 in DataAnalasis.xls ----
### Split the Big Table in Three Tables (PR, JA and News) ----
df <- read.csv("../database/test_bigtable.csv")
news_table = df[df$Source_Category == 'News', c('Sample', 'Reference', 'Source', 'Advice_Code')]
JA_table = df[df$Source == 'Journal Article - Body', c('Reference', 'Design_Actual')]
PR_table = df[df$Source == 'Final Press Release', c('Reference', 'Advice_Code', 'Advice')]

### Create a Table that Crosses Information between JA, PR and News tables ----
# merge the PR Table to the News table
setnames(news_table, 
         old = c('Advice_Code','Source'), 
         new = c('News_Advice_Code','News_Source'))
setnames(PR_table, 
         old = c('Advice_Code','Advice'), 
         new = c('PR_Advice_Code','PR_Exageration'))

d125 <- merge(x = news_table, y = PR_table, by='Reference', all.x = TRUE)
# alternative way, using dplyr:
# d125 <- left_join(news_table, PR_table, by='Reference')

# merge the JA Table to the Result Table:
setnames(JA_table, old = c('Design_Actual'), new = c('JA_Design'))
d125 <- merge(x = d125, y = JA_table, by='Reference', all.x = TRUE)

### Add New Columns To Our Table based on its Current Columns ----
isNewsGreater <- d125$News_Advice_Code > d125$PR_Advice_Code
d125$News_Exageration = ifelse(isNewsGreater, 1, -1)

isNewsEqual = d125$News_Advice_Code == d125$PR_Advice_Code
results_table$News_Exageration[isNewsEqual] <- 0

### Add New Columns based on Older Analysis
sub.d100 <- d100 %>%
  select(Reference, PR_Exageration)

d125 <- left_join(x = d125, y = sub.d100, by='Reference')

# save and display the first 10 rows:
write.csv(d125, "./d125.csv")
head(d125, 10)

File: analyses/analysis125.py
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

df = pd.read_csv("../database/test_bigtable.csv", encoding='utf-8')

news_table = df.ix[df.Source_Category == 'News', ['Sample', 'Reference', 'Source', 'Advice_Code']]
JA_table = df.ix[df.Source == 'Journal Article - Body', ['Reference', 'Design_Actual']]
PR_table = df.ix[df.Source == 'Final Press Release', ['Reference', 'Advice_Code', 'Advice']]

news_table.rename(columns={'Advice_Code': 'News_Advice_Code', "Source": "News_Source"}, inplace=True)
PR_table.rename(columns= {'Advice_Code': 'PR_Advice_Code', 'Advice': 'PR_Exageration'}, inplace=True)

news_table = pd.merge(news_table, PR_table, how='left', on='Reference')

JA_table.rename(columns={'Design_Actual': "JA_Design"})
news_table = pd.merge(news_table, JA_table, how='left', on='Reference')

news_table['News_Exageration'] = np.where(news_table['News_Advice_Code'] - news_table['PR_Advice_Code'] > 0, 1, -1)
when_equality = (news_table['News_Advice_Code'] - news_table['PR_Advice_Code'] == 0)
news_table.ix[when_equality, 'News_Exageration'] = 0

news_table.to_csv('analysis_row125.csv', encoding='utf-8') # never forget the encoding='utf-8'
print "done!"